### PR TITLE
Fix mean() truncating integer and boolean inputs

### PR DESCRIPTION
**Pull request by elrensmin**
## Summary 

`np.mean()` on integer or boolean arrays returned the wrong dtype and value. It computed the mean in float32 but then cast the result back to the input dtype, truncating the fractional part.

```js
np.mean(np.array([1, 2, 3, 4], { dtype: np.int32 }))
// before: int32, 2
// after:  float32, 2.5
```

JAX promotes integer and boolean inputs to float for mean(), and only keeps the original dtype when it is already floating-point (e.g. float16). This change matches that behavior.

## Changes
- src/frontend/core.ts: only cast the mean result back to the original dtype when it is floating-point; leave integer/boolean results as float.
- test/numpy.test.ts: add a jax.numpy.mean() regression suite covering int32  -> float32, bool  -> float32, float32 preserved, and along-axis reduction.

## Verification
- pnpm test test/numpy.test.ts -t "mean" -> 15 passed.
- pnpm test test/numpy.test.ts -> 1159 passed, no new failures.
- pnpm test test/random.test.ts (uses mean) -> 112 passed.
- Verified on both the wasm (default) and cpu backends.
- pnpm lint src/frontend/core.ts test/numpy.test.ts ->  clean.

### CODE REVIEW

**Review by ekzhang — APPROVED** [review]
Thank you! Confirmed behavior with Python, can merge after CI works. Let me know if you have any trouble with that! (`pnpm i && pnpm format`)

```
>>> import jax.numpy as jnp
>>> jnp.mean(jnp.array([1, 2, 3, 4]))
Array(2.5, dtype=float32)
>>> jnp.mean(jnp.array([1, 2, 3, 4], dtype=jnp.float16))
Array(2.5, dtype=float16)
>>> jnp.mean(jnp.array([True, False]))
Array(0.5, dtype=float32)
```

**Comment by elrensmin** [comment]
CI is clear. Ready for merge. Thanks for the quick review!
